### PR TITLE
style: tweak footer layout

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-kstweak-footer-layout-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1203092853539691) 🎟️
- [Figma Link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=16904%3A204196&t=q1A9Dedu7EGodtPd-0) 🎨

## 🗒️ What

This PR tweaks the footer layout according to the new design spec. 
- Adjusts the 'give feedback' link to be styled as a button
- Adjusts the mobile layout to wrap list items
- Adjusts the overall spacing
- Uses css grid to move the 'give feedback' button placement between mobile and desktop

## 📸 Design Screenshots
Screen Shot 2023-01-04 at 2 08 09 PM

See figma link at top

## 🧪 Testing

- [Visit the homepage](https://dev-portal-git-kstweak-footer-layout-hashicorp.vercel.app/), test all scales
- Visit another [content type page](https://dev-portal-git-kstweak-footer-layout-hashicorp.vercel.app/consul/tutorials), test all scales

### CODE REVIEW

**Review by braydenlove — APPROVED** [review]
Love it, awesome work @kendallstrautman. CC @heitzke @kellyschummer

File: src/components/footer/index.tsx [hunk 1]
```diff
@@ -67,8 +63,17 @@ function Footer({
...
 			</a>
+			<ButtonLink
+				text="Give Feedback"
+				href={FEEDBACK_FORM_URL}
+				color="secondary"
+				size="small"
+				icon={<IconPencilTool16 />}
+				opensInNewTab={true}
+				className={s.feedbackButton}
+			/>
```
Comment (ashleemboyer) [inline src/components/footer/index.tsx:74]: 🚧 In the viewport widths where this element visually appears last in the footer, we need to make sure that this element is focused last. Currently, it's always the second focusable element in the `<footer>`. I believe this falls under [SC 2.4.3: Focus Order (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html).

image

We should be able to render this element twice and then use CSS to only show one of them at a time based on the visual order of the elements.

**Review by ashleemboyer — CHANGES_REQUESTED** [review]

File: src/components/footer/index.tsx (hunk 1 above)
Comment (kendallstrautman) [reply src/components/footer/index.tsx:74]: Ah I didn't realize. thank you!

**Comment by braydenlove** [comment]
@kendallstrautman just reviewed with Kelly and Mike and we are all stoked. This is 🚢 able from our end.

File: src/components/footer/index.tsx (hunk 1 above)
Comment (kendallstrautman) [reply src/components/footer/index.tsx]: @ashleemboyer just pushed up the updates based on your notes 😄

Comment (ashleemboyer) [reply src/components/footer/index.tsx]: Taking another look now! 👀

**Comment by kendallstrautman** [comment]
@ashleemboyer oh weird! I will check that out. I'll make a ticket if I don't end up fixing that in this PR!

**Comment by kendallstrautman** [comment]
@ashleemboyer I [added custom focus styles ](https://github.com/hashicorp/dev-portal/pull/1499/commits/0219bb93427f7b67d3df320f6c88701425b2e2b6)based on what I saw in other components. But I didn't see a uniform way we are doing this. Lmk if that approach works!

File: src/components/footer/footer.module.css [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
-	margin: 8px 40px 8px 0;
+	margin: 0 24px var(--logo-button-margin-bottom) 0;
 	position: relative;
+	border-radius: 6px;
+
+	&:focus,
+	&:focus-visible {
+		box-shadow: var(--token-focus-ring-action-box-shadow);
+		outline: transparent;
+	}
```
Comment (ashleemboyer) [inline src/components/footer/footer.module.css:25]: I think what you were looking for is the `g-focus-ring-from-box-shadow` helper. 😊

Can you replace these lines with a `composes` call at the top of the `.logo` block?

```
composes: g-focus-ring-from-box-shadow from global;
```

Comment (kendallstrautman) [reply src/components/footer/footer.module.css]: Thank you! that is the helper I need 😆 much better 👐🏻

**Pull request merged**